src/scheduler/tasks/MigrateSensitiveDataTask.ts

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable @typescript-eslint/indent */
+/* eslint-disable linebreak-style */
+
+import Constants from '../../utils/Constants';
+import Cypher from '../../utils/Cypher';
+import SchedulerTask from '../SchedulerTask';
+import { SettingDB } from '../../types/Setting';
+import SettingStorage from '../../storage/mongodb/SettingStorage';
+import { TaskConfig } from '../../types/TaskConfig';
+import Tenant from '../../types/Tenant';
+
+export default class MigrateSensitiveDataTask extends SchedulerTask {
+
+    public async processTenant(tenant: Tenant, config: TaskConfig): Promise<void> {
+        // Detect if Crypto Key changed
+        await Cypher.detectConfigurationKey(tenant.id);
```

Comment:
It's an initial code sketch but you have to only trigger the migration code path if the key have been changed. 
So returning a boolean here would help to take that decision.